src/main/java/com/artipie/vertx/VertxConnection.java

```diff
@@ -62,13 +62,27 @@
         for (final Map.Entry<String, String> header : headers) {
             this.rsp.putHeader(header.getKey(), header.getValue());
         }
-        this.rsp.setChunked(true);
         final CompletableFuture<HttpServerResponse> promise = new CompletableFuture<>();
-        Flowable.fromPublisher(body)
-            .map(VertxConnection::mapBuffer)
-            .doOnComplete(() -> promise.complete(this.rsp))
-            .doOnError(promise::completeExceptionally)
-            .subscribe(this.rsp.toSubscriber());
+        if (this.rsp.headers().contains("Content-Length")) {
+            this.rsp.setChunked(false);
+            Flowable.fromPublisher(body)
+                .map(VertxConnection::mapBuffer)
+                .doOnComplete(
+                    () -> {
+                        this.rsp.end();
+                        promise.complete(this.rsp);
+                    }
+                )
+                .doOnError(promise::completeExceptionally)
+                .forEach(this.rsp::write);
+        } else {
+            this.rsp.setChunked(true);
+            Flowable.fromPublisher(body)
+                .map(VertxConnection::mapBuffer)
+                .doOnComplete(() -> promise.complete(this.rsp))
+                .doOnError(promise::completeExceptionally)
+                .subscribe(this.rsp.toSubscriber());
+        }
```

Comment:
@Sammers21 I'd suggest to extract common code from these two branches, it would be:
```
Flowable.fromPublisher(body)
  .map(VertxConnection::mapBuffer)
  .doOnError(promise::completeExceptionally)
```